### DISCUSSION TITLE
Autocompaction (smoosh) doesn't cleanup view

### DISCUSSION THREAD

**Discussion by patricks** [apache/couchdb#3103] [Q&A]
Hi everyone,

I am using the official Docker image of CouchDB, v3.1.0.

I currently have the problem that on view gets very big on my disk and never gets compacted.

<img width="271" alt="Screen Shot 2020-08-26 at 09 14 36" src="https://user-images.githubusercontent.com/132167/91272778-929dd680-e77c-11ea-8166-d76856275d0a.png">

I have setup my `couchdb.ini` with the following settings for smoosh (basically the default ones from the docs):

```
[smoosh]
db_channels = upgrade_dbs,ratio_dbs,slack_dbs
view_channels = upgrade_views,ratio_views,slack_views
cleanup_index_files = true

[smoosh.ratio_dbs]
priority = ratio
min_priority = 2.0

[smoosh.ratio_views]
priority = ratio
min_priority = 2.0

[smoosh.slack_dbs]
priority = slack
min_priority = 16777216

[smoosh.slack_views]
priority = slack
min_priority = 16777216
```

Am I doing something wrong or missing something?

Thanks.
 patrick

**Comment by ninetteadhikari** [comment]
Hi there,

It would be great if you can share your log files to give a better understanding of your issue.
A possible compaction error can be `couch_emsort` and if you do see it in your log files then the best approach would be to remove all the `.compact` files and let the compaction rerun. This will resolve the issue. Here is the link to a similar issue reported previously for reference: https://github.com/apache/couchdb/issues/2941

In addition I can see in your Index information the ‘Actual data size’ is only 16. This is unusually small and something you might want to investigate more to ensure your index is correct.

Hope this helps!

**Reply by patricks** [reply]
Hi thanks for your reply,

I have now enabled logging to a file for about 24 hours, log level `info`. But I can't find anything about `emsort`, `compact`, `smoosh`.

There are many entries like this: 
`[notice] 2020-08-28T05:58:02.743534Z nonode@nohost <0.9491.7584> 7f4d088db1 couchdb:5984 X.X.X.X admin GET /db/_design/conflicted/_view/conflicted 200 ok 3`

So it looks like compaction doesn't even run automatically? Maybe I need to enable anything else?

The index of this view still grows.

<img width="268" alt="Screen Shot 2020-08-28 at 08 10 12" src="https://user-images.githubusercontent.com/132167/91527346-ea1b7e00-e905-11ea-8bd5-a98bd8432e15.png">
